pandas/core/apply.py

```diff
@@ -413,11 +413,19 @@ def agg_dict_like(self) -> DataFrame | Series:
                 keys_to_use = ktu
 
             axis: AxisInt = 0 if isinstance(obj, ABCSeries) else 1
-            result = concat(
-                {k: results[k] for k in keys_to_use},
-                axis=axis,
-                keys=keys_to_use,
-            )
+
+            # GH 48581 - Adding consistency to empty agg types
+            if not results:
+                # return empty DataFrame if results is empty, following pattern
+                # empty list and tuple
```

Comment:
I think the comments here can be removed; I think it's clear that the `else` code path will fail in this case and why we're returning an empty frame.